### PR TITLE
inspec compliance version fails gracefully when no server config

### LINKED ISSUE #894: `inspec compliance version` fails with stacktrace if no compliance URL is configured
### Description

The help output lists `inspec compliance version` as a possible command:

```
> inspec help compliance
Commands:
  inspec compliance exec PROFILE    # executes a Chef Compliance profile
  inspec compliance help [COMMAND]  # Describe subcommands or one specific subcommand
  inspec compliance login SERVER    # Log in to a Chef Compliance SERVER
  inspec compliance logout          # user logout from Chef Compliance
  inspec compliance profiles        # list all available profiles in Chef Compliance
  inspec compliance upload PATH     # uploads a local profile to Chef Compliance
  inspec compliance version         # displays the version of the Chef Compliance server
```

However, by default it will yield a stracktrace:

```
> inspec compliance version
/Users/sdanna/.chefdk/gem/ruby/2.1.0/gems/inspec-0.29.0/lib/bundles/inspec-compliance/api.rb:47:in `version': undefined method `+' for nil:NilClass (NoMethodError)
        from /Users/sdanna/.chefdk/gem/ruby/2.1.0/gems/inspec-0.29.0/lib/bundles/inspec-compliance/cli.rb:169:in `version'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:115:in `invoke'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:235:in `block in subcommand'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from /Users/sdanna/.chefdk/gem/ruby/2.1.0/gems/inspec-0.29.0/bin/inspec:9:in `<top (required)>'
        from /Users/sdanna/.chefdk/gem/ruby/2.1.0/bin/inspec:22:in `load'
        from /Users/sdanna/.chefdk/gem/ruby/2.1.0/bin/inspec:22:in `<main>'
```

Based on the code it looks like this is likely a result of not having a compliance server in my configuration.
### InSpec and Platform Version

0.29.0
### Replication Case

`inspec compliance version` without any compliance server configuration.
### Possible Solutions

`inspec` should exit with a non-zero exit status while printing an informative error.

### PR DESCRIPTION
**Pull request by vjeffrey**
screen shot 2016-08-16 at 11 35 13 am

**having trouble running the tests for this at the moment, so i'm putting this up while I work on getting those done.**

fixes https://github.com/chef/inspec/issues/894

### CODE REVIEW

**Comment by chris-rock** [comment]
This needs to be rebased once #928 is merged

**Comment by chris-rock** [comment]
Great work @vjeffrey I added some minor improvements:
- test that the version is returned, once inspec is logged in
- remove whitespace in error message

File: lib/bundles/inspec-compliance/test/integration/default/cli.rb [hunk 1]
```diff
@@ -24,6 +24,13 @@
     its('exit_status') { should eq 0 }
   end
 
+  # version command fails gracefully when server not configured
+  describe command("#{inspec_bin} compliance version") do
+    its('stdout') { should include 'Server configuration information is missing' }
+    its('stderr') { should eq '' }
+    its('exit_status') { should eq 0 }
```
Comment (stevendanna) [inline lib/bundles/inspec-compliance/test/integration/default/cli.rb:31]: Might we want to exit 1 in this case?

**Comment by chris-rock** [comment]
good idea @stevendanna

File: lib/bundles/inspec-compliance/test/integration/default/cli.rb (hunk 1 above)
Comment (chris-rock) [inline lib/bundles/inspec-compliance/test/integration/default/cli.rb:31]: good idea @stevendanna, @vjeffrey could you add a PR to cover that?

**Pull request merged**